### ISSUE TITLE
safe target undoes auth?

### ISSUE THREAD

**Issue by davidpcaldwell** [starkandwayne/safe#189]
I find that if I target a given vault, and then target it again, my `auth` is erased. This is inconvenient when scripting, for two reasons.

1. The most straightforward way to script a safe command, when working with multiple vaults, is to target the vault as part of the script,
2. Having the script be able to produce terminal-based input (like a password) for a subprocess (i.e., `safe`) is difficult in some scripting environments.

The ideal solution for my use case would be to store these authorizations by target so that once authed to a target, one could continue using it, regardless of switching targets.

**Comment by thomasmitchell** [comment]
Hey there,

So, safe _does_ cache tokens for all your targets. If you retarget like `safe target TARGET_NAME`, then it will use the auth token it has cached.

However

If you do `safe target URL TARGET_NAME`, it will think its a new target (even if it targets the same address!) and will wipe the existing entry. I thought I opened an issue for this, but it turns out I just wrote it on my whiteboard. Oops!
